data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -205,14 +212,12 @@ protected boolean shouldKeep(T item) {
       case AVRO:
         return Avro.read(input)
             .project(deleteSchema)
-            .reuseContainers()
```

Comment:
is this related?

Reply:
I think this is needed because records are now placed on a queue inside the parallel iterator. Reusing the record instance instead of copying would cause a problem.

Reply:
This should only be done if the parallelism is greater than 1.